src/planner/explain.rs

```diff
@@ -65,210 +83,261 @@ impl<'a> Explain<'a> {
             costs: self.costs,
             catalog: self.catalog,
             id: *id,
-            depth: self.depth + 1,
         }
     }
 
-    /// Returns a struct displaying the tabs.
-    #[inline]
-    const fn tab(&self) -> impl Display {
-        struct Tab(u8);
-        impl Display for Tab {
-            fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-                for _ in 0..self.0 {
-                    write!(f, "  ")?;
-                }
-                Ok(())
-            }
-        }
-        Tab(self.depth)
-    }
-
     /// Returns a struct displaying the cost.
     #[inline]
-    fn cost(&self) -> impl Display {
-        struct Cost(Option<f32>);
-        impl Display for Cost {
-            fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-                match self.0 {
-                    Some(c) => write!(f, " (cost={c})"),
-                    None => Ok(()),
-                }
-            }
-        }
-        Cost(self.costs.map(|cs| cs[usize::from(self.id)]))
+    fn cost(&self) -> Option<f32> {
+        self.costs.map(|cs| cs[usize::from(self.id)])
     }
 
     /// Returns whether the expression is `true`.
     #[inline]
     fn is_true(&self, id: &Id) -> bool {
         self.expr[*id] == Expr::true_()
     }
-}
 
```

Comment:
```suggestion
    /// Transforms the plan to `Pretty`, an intermediate representation for pretty printing.
```

People may be unfamiliar with the concepts of pretty printing, so it might be better to add some doc comments